engine/src/main/battlecode/world/InternalRobot.java

```diff
@@ -384,8 +384,28 @@ public void empower(int radiusSquared) {
                 conv++;
                 numBotsWithExtraConviction--;
             }
-            // HACK[jerry]: this is the maximum amount the unit can be affected by
-            conv = Math.min(conv, GameConstants.ROBOT_INFLUENCE_LIMIT * 2);
+
+            if (bot.type == RobotType.ENLIGHTENMENT_CENTER && bot.team == this.team) {
+                // conviction doesn't get buffed, do nothing
+            } else if (bot.type == RobotType.ENLIGHTENMENT_CENTER) {
+                // complicated stuff
+                double buff = this.gameWorld.getTeamInfo().getBuff(this.team);
+                long convNeededToConvert = (long) (bot.conviction / buff);
+                while (convNeededToConvert * buff < bot.conviction)
+                    convNeededToConvert++;
+                
+                if (conv < convNeededToConvert) {
+                    // all of conviction is buffed
+                    conv = (long) (conv * this.gameWorld.getTeamInfo().getBuff(this.team));
+                } else {
+                    // conviction buffed until conversion
+                    conv = bot.conviction + (conv - convNeededToConvert);
+                }
```

Comment:
```suggestion
                long convNeededToConvert = (long) (bot.conviction / buff);
                while (convNeededToConvert * buff <= bot.conviction)
                    convNeededToConvert++;
                
                if (conv < convNeededToConvert) {
                    // all of conviction is buffed
                    conv = (long) (conv * this.gameWorld.getTeamInfo().getBuff(this.team));
                } else {
                    // conviction buffed until conversion
                    conv = bot.conviction + 1 - (conv - convNeededToConvert);
                }
```

If the target EC has a conviction of 3, and a politician uses 2 conviction on it with a buff of 2, then the EC will be left at 0 conviction and won't be converted. Not sure if this suggestion is the best way to fix it, please take a look!